applications/FluidDynamicsApplication/custom_utilities/estimate_dt_utilities.h

```diff
@@ -199,39 +199,49 @@ class EstimateDtUtility
         KRATOS_CATCH("")
     }
 
-    /// Calculate each element's CFL for the current time step.
+    /// Calculate each element's CFL for the current time step for the given ModelPart.
     /**
      * The elemental CFL is stored in the CFL_NUMBER elemental variable.
      * To view it in the post-process file, remember to print CFL_NUMBER as a Gauss Point result.
      */
-    void CalculateLocalCFL()
+    static void CalculateLocalCFL(ModelPart& rModelPart)
     {
         KRATOS_TRY;
 
         unsigned int NumThreads = OpenMPUtils::GetNumThreads();
         OpenMPUtils::PartitionVector ElementPartition;
-        OpenMPUtils::DivideInPartitions(mrModelPart.NumberOfElements(),NumThreads,ElementPartition);
+        OpenMPUtils::DivideInPartitions(rModelPart.NumberOfElements(),NumThreads,ElementPartition);
 
-        const double CurrentDt = mrModelPart.GetProcessInfo().GetValue(DELTA_TIME);
+        const double CurrentDt = rModelPart.GetProcessInfo().GetValue(DELTA_TIME);
 
         #pragma omp parallel
         {
             int k = OpenMPUtils::ThisThread();
-            ModelPart::ElementIterator ElemBegin = mrModelPart.ElementsBegin() + ElementPartition[k];
-            ModelPart::ElementIterator ElemEnd = mrModelPart.ElementsBegin() + ElementPartition[k+1];
+            ModelPart::ElementIterator ElemBegin = rModelPart.ElementsBegin() + ElementPartition[k];
+            ModelPart::ElementIterator ElemEnd = rModelPart.ElementsBegin() + ElementPartition[k+1];
```

Comment:
just being curious. Is there any significant improvement between calling DivideInPartitions method or simply doing a loop like:
```
#pragma omp parallel for
for (int ielem = 0; i elem < nelems; ielem++) {
...
}
```
?

Reply:
@ddiezrod I don't think it should make a big difference, if anything your option should be better because it gives openmp more freedom to do things "the right way". This is some old code, from a time we tried to encapsulate openmp calls with the OpenMPUtils class. While that is a good idea in general, this particular method was probably overkill.

Reply:
I see, thanks for the explanation @jcotela !